### PR TITLE
games: /s/name/pname&version/

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/ultimatestunts/default.nix [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {lib, stdenv, fetchurl, SDL, libGLU, libGL, SDL_image, freealut, openal, libvorbis,
 pkg-config}:
 
-stdenv.mkDerivation {
-  name = "ultimate-stunts-0.7.6.1";
+stdenv.mkDerivation rev {
```
Comment (endocrimes) [inline pkgs/games/ultimatestunts/default.nix:4]: /s/rev/rec

**Review by endocrimes — CHANGES_REQUESTED** [review]

File: pkgs/games/xonotic/default.nix [hunk 2]
```diff
@@ -119,8 +119,9 @@ let
 
 in rec {
   xonotic-data = fetchzip {
-    name = "xonotic-data-${version}";
-    url = "https://dl.xonotic.org/${name}.zip";
+    name = "xonotic-data";
+    version = version;
```
Comment (endocrimes) [inline pkgs/games/xonotic/default.nix:123]: fetchzip doesn't take a version parameter (sorry for the 1-by-1 reviews, it's a big pr so I'm finding them mostly by building it)

**Review by endocrimes — CHANGES_REQUESTED** [review]

File: pkgs/games/xonotic/default.nix (hunk 2 above)
Comment (Stunkymonkey) [reply pkgs/games/xonotic/default.nix]: that is fine for me.

File: pkgs/games/rimshot/default.nix [hunk 3]
```diff
@@ -1,9 +1,14 @@
 { lib, stdenv, fetchurl, unzip, love, lua, makeWrapper, makeDesktopItem }:
 
-let
+stdenv.mkDerivation rec {
   pname = "rimshot";
   version = "1.0";
 
+  src = fetchurl {
+    url = "http://stabyourself.net/dl.php?file=${pname}/${pname}-source.zip";
```
Comment (SuperSandro2000) [inline pkgs/games/rimshot/default.nix:8]:
```suggestion
    url = "https://stabyourself.net/dl.php?file=${pname}/${pname}-source.zip";
```

File: pkgs/games/rogue/default.nix [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
+stdenv.mkDerivation rec {
+  pname = "rogue";
+  version = "5.4.4";
 
   src = fetchurl {
     urls = [
-      "https://src.fedoraproject.org/repo/pkgs/rogue/rogue5.4.4-src.tar.gz/033288f46444b06814c81ea69d96e075/rogue5.4.4-src.tar.gz"
+      "https://src.fedoraproject.org/repo/pkgs/rogue/rogue5.4.4-src.tar.gz/033288f46444b06814c81ea69d96e075/rogue${version}-src.tar.gz"
       "http://ftp.vim.org/ftp/pub/ftp/os/Linux/distr/slitaz/sources/packages-cooking/r/rogue5.4.4-src.tar.gz"
       "http://rogue.rogueforge.net/files/rogue5.4/rogue5.4.4-src.tar.gz"
```
Comment (SuperSandro2000) [inline pkgs/games/rogue/default.nix:11]:
```suggestion
      "http://ftp.vim.org/ftp/pub/ftp/os/Linux/distr/slitaz/sources/packages-cooking/r/rogue${version}-src.tar.gz"
      "http://rogue.rogueforge.net/files/rogue${lib.versions.majorMinor version}/rogue${version}-src.tar.gz"
```

File: pkgs/games/rogue/default.nix [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
-stdenv.mkDerivation {
-  name = "rogue-5.4.4";
+stdenv.mkDerivation rec {
+  pname = "rogue";
+  version = "5.4.4";
 
   src = fetchurl {
     urls = [
-      "https://src.fedoraproject.org/repo/pkgs/rogue/rogue5.4.4-src.tar.gz/033288f46444b06814c81ea69d96e075/rogue5.4.4-src.tar.gz"
+      "https://src.fedoraproject.org/repo/pkgs/rogue/rogue5.4.4-src.tar.gz/033288f46444b06814c81ea69d96e075/rogue${version}-src.tar.gz"
```
Comment (SuperSandro2000) [inline pkgs/games/rogue/default.nix:9]:
```suggestion
      "https://src.fedoraproject.org/repo/pkgs/rogue/rogue${version}-src.tar.gz/033288f46444b06814c81ea69d96e075/rogue${version}-src.tar.gz"
```

File: pkgs/games/zdoom/bcc-git.nix [hunk 6]
```diff
@@ -1,7 +1,8 @@
 { lib, stdenv, fetchFromGitHub }:
 
 stdenv.mkDerivation {
-  name = "doom-bcc-git-0.8.0.2018.01.04";
+  pname = "doom-bcc-git";
+  version = "0.8.0.2018.01.04";
```
Comment (SuperSandro2000) [inline pkgs/games/zdoom/bcc-git.nix:5]:
```suggestion
  pname = "doom-bcc";
  version = "unstable-2018-01-04";
```

Comment (Stunkymonkey) [reply pkgs/games/zdoom/bcc-git.nix:5]: could you explain when i should use `unstable`? and why is this not an optional flag?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 127823` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>beret</li>
</ul>
</details>
<details>

  <summary>11 packages failed to build and are new build failures:</summary>
  <ul>
<li>kodiPackages.steam-launcher: log was empty</li>
<li>lutris: log was empty</li>
<li>protontricks: log was empty</li>
<li>steam: log was empty</li>
<li>steam-run: log was empty</li>
<li>steam-tui: log was empty</li>
<li>steamcmd: log was empty</li>
<li>vessel: log was empty</li>
<li>xonotic-dedicated-unwrapped: log was empty</li>
<li>xonotic-glx-unwrapped: log was empty</li>
<li>xonotic-sdl-unwrapped: log was empty</li>
  </ul>
</details>
<details>
<summary>
59 packages built:
</summary>
<ul>
<li>andyetitmoves</li>
<li>asc</li>
<li>ballAndPaddle</li>
<li>bitsnbots</li>
<li>bottles</li>
<li>bsdgames</li>
<li>chessdb</li>
<li>crack_attack</li>
<li>doom-bcc</li>
<li>eboard</li>
<li>eduke32</li>
<li>exult</li>
<li>fsg</li>
<li>gav</li>
<li>gltron</li>
<li>gmad</li>
<li>gnome.atomix</li>
<li>gnome.gnome-klotski</li>
<li>gtetrinet</li>
<li>icbm3d</li>
<li>kobodeluxe</li>
<li>macopix</li>
<li>neverball</li>
<li>njam</li>
<li>openlierox</li>
<li>orthorobot</li>
<li>playonlinux</li>
<li>pokerth</li>
<li>pokerth-server</li>
<li>pong3d</li>
<li>prboom</li>
<li>racer</li>
<li>rili</li>
<li>rimshot</li>
<li>rocksndiamonds</li>
<li>rogue</li>
<li>scrolls</li>
<li>sgtpuzzles</li>
<li>sienna</li>
<li>snake4</li>
<li>steam-run-native</li>
<li>tcl2048</li>
<li>teetertorture</li>
<li>tennix</li>
<li>tintin</li>
<li>tome4</li>
<li>typespeed</li>
<li>uhexen2</li>
<li>ultimatestunts</li>
<li>vectoroids</li>
<li>xbomb</li>
<li>yquake2-all-games</li>
<li>yquake2-ctf</li>
<li>yquake2-ground-zero</li>
<li>yquake2-the-reckoning</li>
<li>zandronum</li>
<li>zandronum-server</li>
<li>zod</li>
<li>zoom</li>
</ul>
</details>

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
This can be merged after xonotic is fixed.

**Comment by Stunkymonkey** [comment]
@SuperSandro2000 anything more to do?

**Comment by SuperSandro2000** [comment]
No. Thank you for your work!

File: pkgs/games/xonotic/default.nix [hunk 7]
```diff
@@ -58,7 +58,7 @@ let
     inherit version;
 
     src = fetchurl {
-      url = "https://dl.xonotic.org/${name}-source.zip";
+      url = "https://dl.xonotic.org/${pname}-${version}-source.zip";
       sha256 = "0axxw04fyz6jlfqd0kp7hdrqa0li31sx1pbipf2j5qp9wvqicsay";
```
[the remark this answers is not in the document]
Comment (Stunkymonkey) [reply pkgs/games/xonotic/default.nix:62]: i just noticed, that your suggestions are not included. i will make a separate PR

**Pull request merged**